yt_dlp/extractor/facebook.py

```diff
@@ -390,7 +390,11 @@ def extract_metadata(webpage):
                 k == 'media' and str(v['id']) == video_id and v['__typename'] == 'Video')), expected_type=dict)
             title = get_first(media, ('title', 'text'))
             description = get_first(media, ('creation_story', 'comet_sections', 'message', 'story', 'message', 'text'))
-            uploader_data = get_first(media, 'owner') or get_first(post, ('node', 'actors', ...)) or {}
+            uploader_data = (
+                get_first(media, ('owner', {dict}))
+                or get_first(post, (..., 'video', lambda k, v: k == 'owner' and v.get('name')), expected_type=dict)
```

Comment:
Filtering in traversal is non-fatal, `.get` is unnecessary
and since we are already filtering for dict keys/values, `expected_type=dict` is not needed:
```suggestion
                or get_first(post, (..., 'video', lambda k, v: k == 'owner' and v['name']))
```
&nbsp;
but now that I think about it, this could probably be simplified to:
```suggestion
                or get_first(post, (..., 'video', 'owner', {dict}))
```
What do you think? Or are there `owner` objects without a `name` key that we specifically need to avoid?

If the latter suggestion works, it could probably be moved up a line

Reply:
Oddly, but yes there are. Example:
```
>>> get_first(post, (..., 'video', 'owner', {dict}))
{'__typename': 'Page', 'id': '316651152345723'}
>>> get_first(post, (..., 'video', lambda k, v: k == 'owner' and v['name']))
{'__typename': 'Page', 'id': '316651152345723', 'name': 'Ls11 Cambodia Gaming Room', '__isVideoOwner': 'Page', 'has_professional_fea..._for_watch': True, 'actor_to_follow': {'__typename': 'Page', '__isActor': 'Page', 'id': '316651152345723', 'video_channel_is_vie..._following': False, 'should_exclude_like_...on_in_feed': False, 'should_show_page_like_warning': False}, 'owner_as_page': {'id': '316651152345723', 'category_type': 'GAMING_VIDEO_CREATOR', 'name': 'Ls11 Cambodia Gaming Room', 'does_viewer_follow': False, 'live_video_subscription_status': 'SOME', 'profile_pic_uri': 'https://scontent-waw...e=647F4EC9', 'woodhenge_creator': None, 'woodhenge_creator_info': None}, '__isActor': 'Page', 'live_video_subscription_status': 'SOME', 'subscribe_status': 'CANNOT_SUBSCRIBE', 'does_viewer_follow': False}
>>> post[1]['video']['owner']
{'__typename': 'Page', 'id': '316651152345723', '__isVideoOwner': 'Page', 'has_professional_fea..._for_watch': True}
>>> post[13]['tahoe_sidepane_renderer']['video']['owner']
{'__typename': 'Page', 'id': '316651152345723', 'name': 'Ls11 Cambodia Gaming Room', '__isVideoOwner': 'Page', 'has_professional_fea..._for_watch': True, 'actor_to_follow': {'__typename': 'Page', '__isActor': 'Page', 'id': '316651152345723', 'video_channel_is_vie..._following': False, 'should_exclude_like_...on_in_feed': False, 'should_show_page_like_warning': False}, 'owner_as_page': {'id': '316651152345723', 'category_type': 'GAMING_VIDEO_CREATOR', 'name': 'Ls11 Cambodia Gaming Room', 'does_viewer_follow': False, 'live_video_subscription_status': 'SOME', 'profile_pic_uri': 'https://scontent-waw...e=647F4EC9', 'woodhenge_creator': None, 'woodhenge_creator_info': None}, '__isActor': 'Page', 'live_video_subscription_status': 'SOME', 'subscribe_status': 'CANNOT_SUBSCRIBE', 'does_viewer_follow': False}
```